File: smarthome.FrontEnd/Smarthome/src/app/models/session.ts
import { UserRole } from './user';

export interface Session {
  token: string;
  userId: number;
  user: {
    userId: number;
    firstName: string;
    lastName: string;
    email: string;
    imageUrl: string;
    password: string;
    role: UserRole;
    registerDate: string;
  };
  expires: string;
}

File: smarthome.Domain/Session.cs
using System.ComponentModel.DataAnnotations;

namespace smarthome.Domain
{
    public class Session
    {
        [Key]
        public string Token { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public User User { get; set; }

        [Required]
        public DateTime Expires { get; set; }
    }
}
